ddtrace/tracer/option.go

```diff
@@ -138,6 +138,13 @@ func WithAgentAddr(addr string) StartOption {
 	}
 }
 
+// WithEnv sets the global environment name. If set, the tracer will tag all
+// traces with "env":"<environment name>". By default, the environment name
+// is not set.
```

Comment:
I don't think the users need to know about the internals of how this works. It should suffice to say:
```go
// WithEnv sets the environment to which all traces started by the tracer will be submitted to.
```